linux_os/guide/system/software/sudo/sudoers_no_root/oval/shared.xml

```diff
@@ -0,0 +1,48 @@
+
+<def-group>
+   <definition class="compliance" id="sudoers_no_root" version="1">
+     {{{ oval_metadata("YOUR SHORT DESCRIPTION HERE") }}}
+     <criteria operator="AND">
+        <criterion comment="Make sure that no rules in sudoers reference root or ALL" test_ref="test_no_roo_or_ALL_rules" />
+        <criterion comment="Make sure that rules in sudoers reference users" test_ref="test_no_user_spec_rules" />
+     </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="none_exist"
+  comment="Make sure that no rules in sudoers reference root or ALL"
+  id="test_no_roo_or_ALL_rules" version="1">
+    <ind:object object_ref="object_root_or_ALL_rules" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_root_or_ALL_rules" version="1">
+    <ind:filepath operation="pattern match">^/etc/sudoers(\.d/.*)?$</ind:filepath>
+    <!-- The regex matches:
+      - The username or group. If root, ignore everything, root can be allowed to do anything.
+      - User specs are present in brackets, and if there are no brackets root user is assumed.
+      - \1: The username or group.
+      - \2: target hostname or ALL
+      - later: word ALL or root inside brackets
+    -->
+    <ind:pattern operation="pattern match">^\s*([\w]+(?&lt;!\broot))\s*(\w+)\s*=\s*(.*,)?\s*\([\w\s]*\b(root|ALL)\b[\w\s]*\)</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="none_exist"
+  comment="Make sure that no rules in sudoers reference root or ALL"
+  id="test_no_user_spec_rules" version="1">
+    <ind:object object_ref="object_no_user_spec_rules" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_no_user_spec_rules" version="1">
+    <ind:filepath operation="pattern match">^/etc/sudoers(\.d/.*)?$</ind:filepath>
+    <!-- The regex matches:
+      - The username or group. If root, ignore everything, root can be allowed to do anything.
+      - User specs are present in brackets, and if there are no brackets root user is assumed.
+      - \1: The username
+      - \2: target hostname or ALL
+      - later: No bracket either right after "hosts =", or no bracket after the "previous command"
+    -->
+    <ind:pattern operation="pattern match">^\s*([\w]+(?&lt;!\broot))\s*(\w+)\s*=\s*(.*,)?\s*[^\(\s]</ind:pattern>
```

Comment:
```suggestion
    <ind:pattern operation="pattern match">^\s*((?!root\b)[\w]+)\s*(\w+)\s*=\s*(.*,)?\s*[^\(\s]</ind:pattern>
```
Same suggestion as above, with negative lookahead, instead of negative lookbehind.